ygnmi/unmarshal.go

```diff
@@ -141,6 +141,9 @@ func unmarshalAndExtract[T any](data []*DataPoint, q AnyQuery[T], goStruct ygot.
 	if !q.isLeaf() {
 		path = proto.Clone(unmarshalledData[0].Path).(*gpb.Path)
 		path.Elem = path.Elem[:len(queryPath.Elem)]
+		if path.Origin != queryPath.Origin {
+			path.Origin = queryPath.Origin
+		}
```

Comment:
this is only really relevant in the batch case where you may paths with differing origins. I didn't want to do unconditionally because I think we should be returning received path in most cases.

Reply:
Is there a difference between doing this unconditionally vs conditionally? Just from the readability perspective I think it's clearer to make this an unconditional assignment and add a comment if this is a notable line. Reading this I would expect complexity to have some purpose, but if the result is the same then I'm having trouble seeing the purpose behind it.

Reply:
sgtm made unconditional